sources/shopify_dlt/__init__.py

```diff
@@ -2,100 +2,134 @@
+    data is limited to items updated in that time range.
+    The range is "half-open", meaning elements equal and newer than `start_time` and elements older than `end_time` are included.
+
     Args:
-        private_app_password (str): The app password to the app on your shop.
-        api_version (str): The API version to use.
-        shop_url (str): The URL of your shop.
-        start_date (Optional[str]): The date from which to import items. It will import items initially created on or after this date.
+        private_app_password: The app password to the app on your shop.
+        api_version: The API version to use (e.g. 2023-01).
+        shop_url: The URL of your shop (e.g. https://my-shop.myshopify.com).
+        per_page: The max number of items to fetch per page. Defaults to 250.
+        start_date: The date from which to import items. It will import items initially created on or after this date.
+            Can be a date/datetime object, an ISO date/datetime string or a pendulum instance. Date/time without timezone is assumed to be UTC.
+        end_time: The end time of the range for which to load data.
+            If end time is not provided, the incremental loading will be enabled and after initial run, only new data will be retrieved
+        order_status: The order status to filter by. Can be 'open', 'closed', 'cancelled', or 'any'. Defaults to 'any'.
 
     Returns:
         Iterable[DltResource]: A list of DltResource objects representing the data resources.
     """
 
     # build client
-    session = shopify.Session(shop_url, api_version, private_app_password)
-    shopify.ShopifyResource.activate_session(session)
+    client = ShopifyApi(shop_url, private_app_password, api_version)
+
+    start_date_obj = ensure_pendulum_datetime(start_date)
+    end_date_obj = ensure_pendulum_datetime(end_date) if end_date else None
 
     # define resources
     @dlt.resource(primary_key="id", write_disposition="merge")
     def products(
-        updated_at: dlt.sources.incremental[str] = dlt.sources.incremental(
-            "updated_at", initial_value=FIRST_DAY_OF_MILLENNIUM
-        )
+        updated_at: dlt.sources.incremental[
+            pendulum.DateTime
+        ] = dlt.sources.incremental(
+            "updated_at", initial_value=start_date_obj, end_value=end_date_obj
+        ),
+        per_page: int = per_page,
     ) -> Iterable[TDataItem]:
         """
         The resource for products on your shop, supports incremental loading and pagination.
 
         Args:
-            updated_at (dlt.sources.incremental[str]): The saved state of the last 'updated_at' value.
+            updated_at: The saved state of the last 'updated_at' value.
 
         Returns:
             Iterable[TDataItem]: A generator of products.
         """
-        page = shopify.Product.find(
-            updated_at_min=updated_at.last_value, created_at_min=start_date
+        params = dict(
```

Comment:
Better if we use a separate `min_created_at` argument in the source I think.     
It's a little problematic with end value loading to use the same start date, because an item's created and updated can each be in range of different chunks so they'd never get loaded even if it should be in the total range.

Reply:
yes you are right. we need to handle backloading differently. 
- when end date is specified we use only `created_at_min` and `created_at_max` to select range
- when we have incremental load we specify `created_at_min` (set to start_date - which is also initial_value) and `updated_at_min` set to last_value
there's a grey are here: should we also track updates of records from backloaded range? in case of the above it is not the case. we'd need another parameter to set created_at and updated_at separately

Reply:
Hmm yeah I'd say it makes sense to load updates from backloaded range too.   
What is the idea behind the created_at cutoff originally?  
I'm struggling with seeing the usefulness in it beyond just filtering by `updated_at`. I'd think old orders you want to ignore aren't updated anyway usually.

Reply:
I think you are always getting the most recent state of the entity and update/create just filter which entities you want to get. so there's no way IMO to get previous state of it. anyway. I kind of agree with what you say.
1. let's have our start/end date settings working only with `updated_at_min` and `updated_at_max` (this one you need to add)
2. I'd add another argument `created_at_min` to cutoff the entities created before that date. the argument is optional and defaults to `FIRST_DAY_OF_MILLENNIUM`

the typical usage would be like in the `incremental_load_with_backloading` example: backloading and then incremental load. all based on updated_at.
On top of that we give possibility to filter out all the record that created before `current_start_date` in the demo. so the `updated_at_min` argument is always set to it, during backloading and on incremental load.

WDYT?

Reply:
That makes a lot of sense, and I think it's more clear with the separate argument.

Reply:
I've added `created_at_min` arg now and test for it.   
`updated_at_max` is used now too.